woocommerce/woocommerce-android
# [Customer Search] Loading, error and empty states

Closes: #9312


### Description

The PR loading, error and empty states to the list

Notice: The UI is not final, but suggestions are welcome

### Testing instructions

* Install nightly WC build https://github.com/woocommerce/woocommerce/releases as the current doesn't have an updated search endpoint
* Using the [plugin](https://github.com/woocommerce/wc-smooth-generator) generate customers for several pages (e.g. 100+)
* Using order creation flow, open the customers screen.
* Search to make the result empty; turn off internet to have an error. 
OR
* Ask me to invite you to the store I used
* Using order creation flow, open the customers screen, scroll down and notice that more customers are loaded
* Search to make the result empty; turn off internet to have an error. 

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4923871/71992e66-d866-4235-aa41-8742f7417005
